hw/ip/spi_device/doc/_index.md

```diff
@@ -412,6 +472,53 @@ instance, if FW sets RXFIFO depth to 128 (default value), it should not update
 the read pointer outside the range 0x000 -  0x1FF (128*4 = 512Bytes ignoring
 the phase bit, bit 11).
 
+## TPM over SPI
+
+### Initialization
+
+The SW should enable the TPM submodule by writing 1 to the TPM_CFG.en CSR field.
+Other SPI_DEVICE features (Generic, Flash, Passthrough) CSRs do not affect the TPM feature.
+
+Update TPM_ACCESS_0, TPM_ACCESS_1 CSRs.
+The TPM submodule uses TPM_ACCESS_x.activeLocality to determine if the TPM_STS is returned to the SB.
+The SW may configure TPM_CFG.hw_reg_dis and/or TPM_CFG.invalid_locality to fully control the TPM transactions.
+
+### TPM mode: FIFO and CRB
+
+The HW returns the return-by-HW registers for the read request in the TPM FIFO mode.
+In the TPM CRB mode (TPM_CFG.tpm_mode is 1), the logic always upload the command and address to the SW and waits for the read FIFO data even the received address falls into the managed address.
+
+### Return-by-HW register update
+
+The SW manages the retun-by-HW registers.
+The contents are placed inside the SPI_DEVICE CSRs.
+The SW must maintain the other TPM registers outside of the SPI_DEVICE HWIP and uses write/read FIFOs to receive the content from/ send the register value to the SB.
+
+When the SW updates the return-by-HW registers, the SW is recommended to read back the register to confirm the value is written.
+Due to the CDC issue, the SW is permitted to update the registers when the TPM CS# is de-asserted.
+
+### TPM Read
+
+1. The SB sends the TPM read command with the address.
+1. The SW reads a word from TPM_CMD_ADDR CSR (optional cmdaddr_notempty interrupt).
+  1. If the address falls into the return-by-HW registers and TPM_CFG.hw_reg_dis is not set, the HW does not push the command and address bytes into the TPM_CMD_ADDR CSR.
+1. The SW prepares the register value and writes the value into the read FIFO.
+1. The TPM submodule sends `WAIT` until the read FIFO is available. When available, the TPM submodule sends `START` followed by the register value.
```

Comment:
this just occurred to me, if in the future we become more than 4B, do we need to add a "trigger" of some kind? Because once we start, we cannot wait again.. but if the host side is too slow at filling up more than 4B of data?

Reply:
As transfer size is 64B, I assume SW always has the read data on its hands. If we ever have to support 64B with current IP, then the SW should keep checking the read fifo depth and push more bytes before the FIFO becomes empty. I think that time will not come soon. We may increase the read FIFO depth if we re-spin the chip

Reply:
o yes, what i meant was this...software pushes 4B in... the hardware begins withdrawing... for whatever reason, software is interrupted, so it does not push the next 4B in, then when hardware finishes pushing out the 4B, it will end up transferring empty data. 

I think it might be reasonable to assume that the data pushing MIGHT not be done in an interrupt routine, so the software could potentially jump away.